### PR TITLE
Add update, default and destroy loan scheme calls

### PR DESCRIPTION
**Pull request by Bushstar**
Add update, default and destroy loan scheme calls as specified in the pink paper on loans.

https://github.com/DeFiCh/pinkpaper/tree/main/loan

### CODE REVIEW

File: src/masternodes/rpc_loan.cpp [hunk 1]
```diff
@@ -341,7 +413,177 @@ UniValue createloanscheme(const JSONRPCRequest& request) {
...
+        execTestTx(CTransaction(rawTx), targetHeight, metadata, CLoanSchemeMessage{}, coinview);
+    }
+
+    return signsend(rawTx, pwallet, optAuthTx)->GetHash().GetHex();
+}
+
+UniValue setdefaultloanscheme(const JSONRPCRequest& request) {
+    CWallet* const pwallet = GetWallet(request);
+
+    RPCHelpMan{"createloanscheme",
```
Comment (hidiego) [inline src/masternodes/rpc_loan.cpp:425]:
```suggestion
    RPCHelpMan{"setdefaultloanscheme",
```

**Comment by Bushstar** [comment]
To-do: On destroy delete all pending updates.

**Review by hidiego — APPROVED** [review]

File: src/masternodes/mn_checks.cpp [hunk 2]
```diff
@@ -1665,48 +1679,139 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
...
 
-    Res operator()(const CCreateLoanSchemeMessage& obj) const {
+    Res operator()(const CLoanSchemeMessage& obj) const {
         if (!HasFoundationAuth()) {
             return Res::Err("tx not from foundation member!");
         }
 
         if (obj.ratio < 100) {
-            return Res::Err("Ratio cannot be less than 100");
+            return Res::Err("mincolratio cannot be less than 100");
```
Comment (ShengguangXiao) [inline src/masternodes/mn_checks.cpp:1688]: Error message use fullname will be more clear, like "minimum collateral ratio"

File: src/masternodes/mn_checks.cpp [hunk 3]
```diff
@@ -1665,48 +1679,139 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
...
         }
 
         if (obj.rate < 1000000) {
-            return Res::Err("Rate cannot be less than 0.01");
+            return Res::Err("interestrate cannot be less than 0.01");
```
Comment (ShengguangXiao) [inline src/masternodes/mn_checks.cpp:1692]: should use "interest rate"

**Review by bvbfan — APPROVED** [review]

**Pull request merged**